lib/http/preprocessors.js

```diff
@@ -122,8 +122,11 @@ const authHandler = ({ Sessions, Users, Auth }, context) => {
       // if authentication failed anyway, just do nothing.
       if ((cxt == null) || cxt.auth.session.isEmpty()) return;
 
-      const csrf = urlDecode(cxt.body.__csrf);
-      if (csrf.isEmpty() || isBlank(csrf.get()) || (cxt.auth.session.get().csrf !== csrf.get())) {
+      // Neither csrf nor x-requested-with provided for POST request, so reject it.
+      const csrf = urlDecode(cxt.body?.__csrf);
+      if (context.method === 'POST' &&
```

Comment:
We already check `context.method` on line 118 above. What do you think about modifying that line instead of also checking `context.method` here?